### ISSUE TITLE
SslProtocols obsolete warnings-as-errors while trying to upgrade to newest packages

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#5708]
e.g. http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/windows_nt_release_prtest/1441/
```
d:\j\workspace\dotnet_corefx\windows_nt_debug_prtest\src\Common\tests\System\Net\SslProtocolSupport.cs(21,15): error CS0618: 'SslProtocols.Ssl3' is obsolete: 'This value has been deprecated.  It is no longer supported. http://go.microsoft.com/fwlink/?linkid=14202' [d:\j\workspace\dotnet_corefx\windows_nt_debug_prtest\src\System.Net.Http.WinHttpHandler\tests\UnitTests\System.Net.Http.WinHttpHandler.Unit.Tests.csproj]
```
cc: @davidsh, @CIPop, @SidharthNabar

**Comment by davidsh** [comment]
This looks like a WinHttpHandler UnitTest that needs to be removed since it was using Ssl30 and we already removed support for that.

**Comment by davidsh** [comment]
Actually, this is more complicated.  And I don't understand why we didn't see this in CI before #4483 was merged.

The code above is failing to compile the WinHttpHandler UnitTests due to code in the Common file SslProtocolSupport:

```c#
        public const SslProtocols UnsupportedSslProtocols =
            SslProtocols.Ssl2
            | SslProtocols.Ssl3;
```

**Comment by stephentoub** [comment]
> I don't understand why we didn't see this in CI before #4483 was merged.

Because the project depends on the package, and it's pulling a specific version of the package, not compiling against the live code in the other project.
cc: @weshaggard

**Comment by CIPop** [comment]
> Because the project depends on the package, and it's pulling a specific version of the package, not compiling against the live code in the other project.

+1 Was about to hit enter on my reply :).
@stephentoub and I were discussing offline about moving all System.Net tests to follow this pattern this exact pattern yesterday. 
I'll open a tracking issue for the TODOs in most of the System.Net contract implementations and separate issues for all other System.Net contracts to ensure this doesn't happen again, at least within the System.Net owned code.

Working on a fix involving [#pragma warning](https://msdn.microsoft.com/en-us/library/441722ys.aspx) disable.

**Comment by davidsh** [comment]
Would a developer have seen this if they had done this before it got merged:

`msbuild /t:rebuildandtest /p:uselocallibraries=true`

**Comment by stephentoub** [comment]
> Would a developer have seen this if they had done this before it got merged

Theoretically it would have caught the problem.  However, I believe @weshaggard is looking into a potential bug with that flag. cc: @ellismg

**Comment by CIPop** [comment]
>> Would a developer have seen this if they had done this before it got merged

> Theoretically it would have caught the problem. However, I believe @weshaggard is looking into a potential bug with that flag. cc: @ellismg 

It doesn't and I can't repro this in my environment. Trying running `clean && build`

**Comment by davidsh** [comment]
Reference PR that failed, #5703.

**Comment by stephentoub** [comment]
> I can't repro this in my environment

The PR that failed was one trying to update to the newer packages.  You should be able to repro if you pull that PR down locally:
```
git fetch upstream pull/5703/head:update_packages
git checkout update_packages
```

**Comment by CIPop** [comment]
Thanks @stephentoub! I have a fix but I'm not sure of the easiest way to integrate with the original PR.

**Comment by stephentoub** [comment]
You could either merge your fix once the CI is green, and then re-run the other PR, or @Priya91 can cherry-pick your commit into her PR.

**Comment by weshaggard** [comment]
I'm looking at the uselocallibraries issue as it is currently broken for a number of scenarios. 

>separate issues for all other System.Net contracts to ensure this doesn't happen again, at least within the System.Net owned code.

@CIPop I'm not sure how you plan to change this but please cc me on the changes because we have to be careful about how we change this as we cannot build against all live things.

**Comment by CIPop** [comment]
@weshaggard I plan to change _test_ csproj files for System.Net tests to point to other System.Net contracts. We're already doing this for some of the contracts:

https://github.com/dotnet/corefx/blob/master/src/System.Net.Sockets/tests/FunctionalTests/System.Net.Sockets.Tests.csproj#L77

Closing this issue as it has been resolved by 6609494 which has been merged in #5703.
